src/gui/wallet_test.go

```diff
@@ -548,10 +555,191 @@ func TestWalletGet(t *testing.T) {
-			}
+			require.NoError(t, err)
 			require.Equal(t, tc.gatewayGetWalletResult, msg, tc.name)
 		}
 	}
 }
+
+func TestWalletNewAddressesHandler(t *testing.T) {
+	type httpBody struct {
+		Id  string
+		Num string
+	}
+	type Addresses struct {
+		Address []string `json:"addresses"`
+	}
+	var responseAddresses = Addresses{}
+	var responseEmptyAddresses = Addresses{}
+
+	var emptyAddrs = make([]cipher.Address, 0)
+	var addrs = make([]cipher.Address, 3)
+
+	for i := 0; i < 3; i++ {
+		pub, _ := cipher.GenerateDeterministicKeyPair(cipher.RandByte(32))
+
+		addrs[i] = cipher.AddressFromPubKey(pub)
+		responseAddresses.Address = append(responseAddresses.Address, addrs[i].String())
+	}
+	tt := []struct {
+		name                      string
+		method                    string
+		url                       string
+		body                      *httpBody
+		status                    int
+		err                       string
+		walletId                  string
+		n                         uint64
+		gatewayNewAddressesResult []cipher.Address
+		gatewayNewAddressesErr    error
+		responseBody              Addresses
+	}{
+		{
+			"405",
+			http.MethodPut,
+			"/wallet/create",
+			nil,
+			http.StatusMethodNotAllowed,
+			"405 Method Not Allowed",
+			"",
+			1,
+			make([]cipher.Address, 0),
+			nil,
+			Addresses{},
+		},
+		{
+			"400 - missing wallet id",
+			http.MethodPost,
+			"/wallet/create",
+			nil,
+			http.StatusBadRequest,
+			"400 Bad Request - missing wallet id",
+			"foo",
+			1,
+			make([]cipher.Address, 0),
+			nil,
+			Addresses{},
+		},
+		{
+			"400 - invalid num value",
+			http.MethodPost,
+			"/wallet/create",
+			&httpBody{
+				Id:  "foo",
+				Num: "bar",
+			},
+			http.StatusBadRequest,
+			"400 Bad Request - invalid num value",
+			"foo",
+			1,
+			make([]cipher.Address, 0),
+			nil,
+			Addresses{},
+		},
+		{
+			"400 - gateway.NewAddresses error",
+			http.MethodPost,
+			"/wallet/create",
+			&httpBody{
+				Id:  "foo",
+				Num: "1",
+			},
+			http.StatusBadRequest,
+			"400 Bad Request - gateway.NewAddresses error",
+			"foo",
+			1,
+			make([]cipher.Address, 0),
+			errors.New("gateway.NewAddresses error"),
+			Addresses{},
+		},
+		{
+			"400 - gateway.NewAddresses error",
+			http.MethodPost,
+			"/wallet/create",
+			&httpBody{
+				Id:  "foo",
+				Num: "1",
+			},
+			http.StatusBadRequest,
+			"400 Bad Request - gateway.NewAddresses error",
+			"foo",
+			1,
+			make([]cipher.Address, 0),
+			errors.New("gateway.NewAddresses error"),
+			Addresses{},
+		},
+		{
+			"200 - OK",
+			http.MethodPost,
+			"/wallet/create",
+			&httpBody{
+				Id:  "foo",
+				Num: "1",
+			},
+			http.StatusOK,
+			"",
+			"foo",
+			1,
+			addrs,
+			nil,
+			responseAddresses,
+		},
+		{
+			"200 - OK empty addresses",
+			http.MethodPost,
+			"/wallet/create",
+			&httpBody{
+				Id:  "foo",
+				Num: "1",
+			},
+			http.StatusOK,
+			"",
+			"foo",
+			1,
+			emptyAddrs,
+			nil,
+			responseEmptyAddresses,
```

Comment:
what is this case testing?

Reply:
This case shows that the result of gateway.NewAddresses can be an empty []cipher.Address, but the response status is 200.

Reply:
ok, only when num is 0

Reply:
it should only return empty addrs if num is 0

Reply:
Changed 1 -> 0